app/2.0/docs/tools/build-for-production.md

```diff
@@ -0,0 +1,222 @@
+---
+title: Build for production
+---
+
+<!-- toc -->
+
+## Overview
+
+### Why do we need  a build process?
+
+The tools in the Polymer build process can optimize your application. The build toolset can reduce the download size of your app and its elements by minifying your code. By concatenating your app's element definitions, scripts and stylesheets, the build toolset can also reduce the number of network requests a user's browser will make when loading your app and its resources. 
+
+The Polymer build tools ensure that your applications can run across multiple and older browsers. Polymer 2.0 is written in ES6, for more legible, maintainable, and modular code. For compatibility with older browsers, the Polymer build tools must compile ES6 to ES5. 
+
+Some application developers will need to deliver JavaScript separately from HTML files - for example, for deployment to environments that use [CSP (Content Security Policy)](https://developers.google.com/web/fundamentals/security/csp/). The Polymer build tools can handle this requirement by splitting inline JavaScript and CSS from HTML.
+
+### What build tools are available?
+
+* If your application is written entirely in Polymer and does not use any other tools or frameworks, you can simply use the Polymer CLI to build your application:
+
+  ```
+  `polymer build`
+  ```
+
+  You can configure your build, or create multiple builds, by writing a [`polymer.json`](https://www.polymer-project.org/2.0/docs/tools/polymer-json) configuration file.
+  
+* If you need a custom build process, for example, to integrate another tool into your build chain, you can [use the `polymer-build` library](#use-polymer-build) to do so. 
+
+## Understanding the transforms
+
+The Polymer CLI and `polymer-build` library support the following transforms:
+
+* [Splitting inline JavaScript and CSS from HTML files](#splitting)
+* [Minifying HTML, JavaScript and CSS](#minifying)
+* [Compiling ES6 to ES5](#compiling)
+* [Bundling resources to reduce the total number of HTTP requests made by the user's browser](#bundling)
+
+### Splitting inline JavaScript and CSS from HTML files {#splitting}
+  
+By default, the Polymer CLI build process uses a built in feature of `polymer-build` to split inline JavaScript and CSS from HTML files. The build process automatically splits these streams in order to process them appropriately - for example, for minification, compilation or both.
```

Comment:
I have left this out for now. It looks like the most helpful thing to do would be to include a section on CSP per the old docs. I created https://github.com/Polymer/docs/issues/2101 to track this.